elasticsearch-curator: init at 5.4.1

basvandijk opened the pull request:
###### Motivation for this change

This is a handy tool for tending your Elasticsearch indices.

See: https://github.com/elastic/curator

###### Things done

For meeting dependencies I added the `voluptuous` Python library and upgraded the `elasticsearch` Python library from 1.9.0 to 6.0.0.

I tested this by running: 

`curator_cli --host my-elasticsearch-host show_indices`

and observing the expected list of indici.

Note that this is both a Python library as well as a Python CLI script.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

orivej commented:
Could you split the changes to different packages into different commits, delete `pyelasticsearch` because it requires ancient version of `elasticsearch` which may not be compatible with any existing installations, and add missing `nosexcover` (and possibly other) check input(s) to `elasticsearch-curator`.

basvandijk committed:
python.pkgs.voluptuous: init at 0.10.5

Needed by elasticsearch-curator.

basvandijk committed:
python.pkgs.elasticsearch: 1.9.0 -> 6.0.0

basvandijk committed:
python.pkgs.elasticsearch-curator: init at 5.4.1

basvandijk commented:
@orivej thank you for reviewing this! I made your suggested changes. The package now builds with `--option build-use-sandbox true`.

orivej committed:
python.pkgs.pyelasticsearch: delete

because it requires python.pkgs.elasticsearch 1.x.y, which is compatible only
with the ancient Elasticsearch 1.

orivej commented:
Thanks!
